FT2-109-Advanced: Completed the Advanced SQL assignment.

MD-Aftab-Innoraft opened the pull request:
- Added proper comments for functions and variables.
- Added front-end validations.

MD-Aftab-Innoraft committed:
FT2-109-Advanced: Completed SQL Advanced assignment.
Description:
- Created a login/signup system in PHP.
- Created feature for resetting password.
- Creating a pdf based on user inputs.

MD-Aftab-Innoraft committed:
FT2-109-Advanced: Improved appearance, eased navigation, corrected validations.
Description:
- Added CSS for signup, reset password and login pages.
- Corrected the front-end JS validation for form inputs.
- Used environment variables to hide sensitive database credentials.
- Improved navigation between home,forgot password, reset password pages.

Md-Kaif-Innoraft commented on .env.example:
```diff
@@ -0,0 +1,4 @@
+DB_HOST =
+DB_USER =
+DB_PASS =
+DB_DATABASE =
```
Why are you sending .env file on github.

MD-Aftab-Innoraft replied:
It is an .env.example file to let anyone know what environment variables are used.

Md-Kaif-Innoraft commented on Form.php:
```diff
@@ -0,0 +1,357 @@
+   */
+  private $indianPhoneNumber;
+
+
```
why is here an extra line?

Md-Kaif-Innoraft commented on Form.php:
```diff
@@ -0,0 +1,357 @@
+    // No error.
+    return "";
+  }
+
```
In the above method pls write the explained comment.

Md-Kaif-Innoraft commented on Form.php:
```diff
@@ -0,0 +1,357 @@
+      $this->status = TRUE;
+    }
+    return $this->status;
+  }
```
Correct the indentation.

Md-Kaif-Innoraft commented on Form.php:
```diff
@@ -0,0 +1,357 @@
+    }
+    else {
+      // Initialising curl.
+      $ch = curl_init();
```
Why don't  we using guzzlehttp?

MD-Aftab-Innoraft replied:
I wanted to explore curl as an alternative to guzzlehttp/guzzle.

Md-Kaif-Innoraft commented on Form.php:
```diff
@@ -0,0 +1,357 @@
+      // Initialising curl.
+      $ch = curl_init();
+      // Attaching email address to be verified.
+      curl_setopt($ch, CURLOPT_URL, 'https://emailvalidation.abstractapi.com/v1/?api_key=46cc4fa558dd4a17b2fbbd8402620c2e&email='. $email);
```
why we are providing api key here?

Md-Kaif-Innoraft commented on database.php:
```diff
@@ -0,0 +1,28 @@
+$dotenv->load();
+
+// Establishing connection with MySQL.
+$mysqli = new mysqli(hostname: $_ENV['DB_HOST'],
```
why don't we are using pdo?

Md-Kaif-Innoraft commented on database.php:
```diff
@@ -0,0 +1,28 @@
+// Returning the connection object.
+return $mysqli;
+
+?>
```
Not required.

Md-Kaif-Innoraft commented on logout.php:
```diff
@@ -0,0 +1,15 @@
+header("location: index.php");
+exit;
+
+?>
```
Not required.

MD-Aftab-Innoraft replied:
May be it's not necessary but I came across a tutorial on youtube in which the host said that it's a good practice to redirect the user to the index(login/signup) page after destroying the session.

Md-Kaif-Innoraft commented on mailer.php:
```diff
@@ -0,0 +1,28 @@
+// Returning the configured PHP mailer object.
+return $mail;
+
+?>
```
Not required.

MD-Aftab-Innoraft committed:
FT2-109-Advanced: Made corrections as per PR review.
Description:
- Removed ?

MD-Aftab-Innoraft committed:
FT2-109-Advanced: Made relevant changes as per PR review.
Description:
- Removed '?>' from the end of files containing only PHP code.
- Removed the API key from the pushed code and stored in the .env file.
- Corrected indentation and removed in-appropriate spacing at few locations in code.

MD-Aftab-Innoraft committed:
FT2-109-Advanced: Changed naming convention of files.
Description:
- Changed the naming style of files from kebab-case to camel-case.

Md-Kaif-Innoraft reviewed (approved):
Satisfied by the changes.
